### PR TITLE
JCLOUDS-813: fix list private images

### PR DESCRIPTION
**Pull request by andreaturli**
To fix the bug reported at JCLOUDS-813 this PR revisit the strategy to listImages:
- modify the listImages and getImage strategies in SoftLayerComputeServiceAdapter: by default, it only consider `stock` images
- add support for VirtualGuestBlockDeviceTemplateGroupApi getObject: this is need in `getImage` if the user specify a specific `imageId` to scan public and private images + stock
- remove PROPERTY_SO FTLAYER_INCLUDE_PUBLIC_IMAGES: this is no more needed

### CODE REVIEW

**Comment by andreaturli** [comment]
@nacx can you have another look at it? thanks!

File: providers/softlayer/src/main/java/org/jclouds/softlayer/features/VirtualGuestBlockDeviceTemplateGroupApi.java [hunk 1]
```diff
@@ -51,8 +52,17 @@
...
 
+   /**
+    * @return public images
+    * @see <a href="http://sldn.softlayer.com/reference/services/SoftLayer_Virtual_Guest_Block_Device_Template_Group/getObject" />
+    */
+   @Named("VirtualGuestBlockDeviceTemplateGroup:getObject")
+   @GET
+   @Path("/SoftLayer_Virtual_Guest_Block_Device_Template_Group/{id}/getObject")
+   @QueryParams(keys = "objectMask", values = MASK)
+   VirtualGuestBlockDeviceTemplateGroup getObject(@PathParam("id") String id);
```
Comment (nacx) [inline providers/softlayer/src/main/java/org/jclouds/softlayer/features/VirtualGuestBlockDeviceTemplateGroupApi.java]: Add the corresponding unit and live tests for this new api method.

File: providers/softlayer/src/main/java/org/jclouds/softlayer/features/VirtualGuestBlockDeviceTemplateGroupApi.java [hunk 2]
```diff
@@ -51,8 +52,17 @@
    @Named("VirtualGuestBlockDeviceTemplateGroup:getPublicImages")
    @GET
    @Path("/SoftLayer_Virtual_Guest_Block_Device_Template_Group/getPublicImages")
-   @QueryParams(keys = "objectMask", values = LIST_PUBLIC_IMAGES_MASK)
+   @QueryParams(keys = "objectMask", values = MASK)
    @Fallback(Fallbacks.EmptySetOnNotFoundOr404.class)
    Set<VirtualGuestBlockDeviceTemplateGroup> getPublicImages();
 
+   /**
+    * @return public images
```
Comment (nacx) [inline providers/softlayer/src/main/java/org/jclouds/softlayer/features/VirtualGuestBlockDeviceTemplateGroupApi.java]: This returns a single image.

**Comment by andreaturli** [comment]
@nacx I've addressed your comments, sorry I forgot about the tests.

I need to port the expectTests to MWS but not in this PR

**Comment by andreaturli** [comment]
@nacx if the builder is happy can backport this to 1.8.x as well?

**Comment by nacx** [comment]
Feel free to backport it to 1.8.x, but it is unlikely that another 1.8 version is released.

**Comment by andreaturli** [comment]
Great, @nacx! The builder is happy, can I merge it now?

**Comment by andreaturli** [comment]
Great, @nacx! Squashed and rebased but the builder is not happy, because of an unrelated issue. Closing and re-opening the PR to trigger the builder again

**Comment by nacx** [comment]
+1 once the build is green again. Thx @andreaturli !

**Pull request merged**